File: app/src/main/java/com/bestxty/sault/demo/DownloadApplication.java
package com.bestxty.sault.demo;

import android.app.Application;
import android.content.Context;
import android.os.Environment;

import com.bestxty.sault.Sault;
import com.bestxty.sault.SaultConfiguration;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

import java.io.File;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * @author xty
 *         Created by xty on 2016/11/29.
 */
public class DownloadApplication extends Application {

    private RefWatcher mRefWatcher;

    @Override
    public void onCreate() {
        super.onCreate();

        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }

        mRefWatcher = LeakCanary.install(this);
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.HEADERS);


        SaultConfiguration saultConfiguration = new SaultConfiguration.Builder()
                .saveDir(Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator + "dl_test")
                .client(new OkHttpClient.Builder()
                        .addInterceptor(loggingInterceptor)
                        .build())
                .loggingEnabled(true)
                .build();

        Sault.setDefaultConfiguration(saultConfiguration);


    }

    public static RefWatcher getRefWatcher(Context context) {
        return ((DownloadApplication) context.getApplicationContext()).mRefWatcher;
    }
}

File: app/src/main/java/com/bestxty/sault/demo/DownloadApplication.kt
package com.bestxty.sault.demo

import android.app.Application
import android.content.Context
import android.os.Environment
import com.bestxty.sault.Sault
import com.bestxty.sault.SaultConfiguration
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.io.File

/**
 * @author xty
 *         Created by xty on 2016/11/29.
 */
class DownloadApplication : Application() {

    lateinit var mRefWatcher: RefWatcher

    override fun onCreate() {
        super.onCreate()
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return
        }
        mRefWatcher = LeakCanary.install(this)
        val loggingInterceptor = HttpLoggingInterceptor()
        loggingInterceptor.level = HttpLoggingInterceptor.Level.HEADERS
        val saultConfiguration: SaultConfiguration = SaultConfiguration.Builder()
                .saveDir(Environment.getExternalStorageDirectory()
                        .absolutePath + File.separator + "sault_test")
                .client(OkHttpClient.Builder()
                        .addInterceptor(loggingInterceptor)
                        .build())
                .build()
        Sault.setDefaultConfiguration(saultConfiguration)

    }


    companion object {
        fun getRefWatcher(context: Context): RefWatcher {
            val downloadApplication = context.applicationContext as DownloadApplication
            return downloadApplication.mRefWatcher
        }
    }

}
